src/Travers.php

```diff
@@ -65,6 +88,40 @@ public function find($key)
         return $this->parseParam($this->parseKeys($key), $this->tree);
     }
 
+    /**
+     * @param       $key
+     * @param array $tree
+     *
+     * @return array|mixed
+     *
+     * @throws BranchNotFoundException
+     */
+    protected function removeParam($key)
```

Comment:
let's have this function take the tree as an argument instead of using $this->tree